x-pack/plugins/cloud/server/plugin.ts

```diff
@@ -50,6 +50,10 @@ export class CloudPlugin implements Plugin<CloudSetup> {
     registerCloudDeploymentIdAnalyticsContext(core.analytics, this.config.id);
     registerCloudUsageCollector(usageCollection, { isCloudEnabled });
 
+    if (isCloudEnabled) {
+      security?.setIsElasticCloudDeployment();
+    }
```

Comment:
NIT/optional: Shame on us, we don't yet have a test file for `x-pack/plugins/cloud/server/plugin.ts`, but it would make sense to add one to test this call/behavior

Reply:
That's a good point, I'll add one.

Reply:
Added few basic tests in https://github.com/elastic/kibana/pull/132622/commits/c26c503bc0928543f75bdd32c2bd2188e75f2bb5